windows/src/engine/keyman32/appint/aiTIP.cpp

```diff
@@ -595,7 +595,6 @@ BOOL AITIP::QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR
 {
   PKEYMAN64THREADDATA _td = ThreadGlobals();
   if(!_td) return TRUE;
-  if(!_td->ForceFileName[0]) return TRUE;
```

Comment:
QueueDebugInformation,
Before removing ForceFileName this would return early if it wasn't a force filename. The onion layers continue. How do we "switch off"  `QueueDebugInformation`

Reply:
I reckon you'll find that `QueueDebugInformation` should be unused now that we've stripped out the legacy core. Any remaining references will be hints to additional code we should be able to strip out.

```
21 results - 6 files

windows\src\engine\keyman32\kmprocess.cpp:
  169  		if(_td->app->IsUnicode())
  170: 			_td->app->QueueDebugInformation(QID_BEGIN_UNICODE, NULL, NULL, NULL, NULL, (DWORD_PTR) &keyinfo);
  171  		else
  172: 			_td->app->QueueDebugInformation(QID_BEGIN_ANSI, NULL, NULL, NULL, NULL, (DWORD_PTR) &keyinfo);
  173  	}

  258  
  259: 	_td->app->QueueDebugInformation(QID_END, NULL, NULL, NULL, NULL, 0);
  260  	return !fOutputKeystroke;

  296  
  297: 	_td->app->QueueDebugInformation(QID_GROUP_ENTER, gp, NULL, NULL, NULL, 0);
  298  

  313  		_td->state.StopOutput = TRUE;
  314: 		_td->app->QueueDebugInformation(QID_GROUP_EXIT, gp, NULL, NULL, NULL, QID_FLAG_RECURSIVE_OVERFLOW);
  315  		return FALSE;

  459  			/* NoMatch rule found, and is a character key */
  460: 			_td->app->QueueDebugInformation(QID_NOMATCH_ENTER, gp, NULL, NULL, gp->dpNoMatch, 0);
  461  			PostString(gp->dpNoMatch, &_td->state.msg, _td->state.lpkb, NULL);
  462: 			_td->app->QueueDebugInformation(QID_NOMATCH_EXIT, gp, NULL, NULL, gp->dpNoMatch, 0);
  463  		}

  472  
  473: 		_td->app->QueueDebugInformation(QID_GROUP_EXIT, gp, NULL, NULL, NULL, QID_FLAG_NOMATCH);
  474  		return TRUE;

  500  
  501: 	_td->app->QueueDebugInformation(QID_RULE_ENTER, gp, kkp, _td->miniContext, NULL, 0);
  502  

  544  	{
  545: 		_td->app->QueueDebugInformation(QID_RULE_EXIT, gp, kkp, _td->miniContext, NULL, 0);
  546  

  548  		{
  549: 			_td->app->QueueDebugInformation(QID_MATCH_ENTER, gp, NULL, NULL, gp->dpMatch, 0);
  550  			PostString(gp->dpMatch, &_td->state.msg, _td->state.lpkb, NULL);
  551: 			_td->app->QueueDebugInformation(QID_MATCH_EXIT, gp, NULL, NULL, gp->dpMatch, 0);
  552  		}

  554  	else
  555: 		_td->app->QueueDebugInformation(QID_RULE_EXIT, gp, kkp, _td->miniContext, NULL, 0);
  556: 	_td->app->QueueDebugInformation(QID_GROUP_EXIT, gp, NULL, NULL, NULL, 0);
  557  

windows\src\engine\keyman32\appint\aiTIP.cpp:
  725  
  726: BOOL AITIP::QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR fcontext, PWSTR foutput, DWORD_PTR dwExtraFlags)
  727  {

  731  
  732: 	SendDebugMessageFormat(0, sdmAIDefault, 0, "AIDebugger::QueueDebugInformation ItemType=%d", ItemType);
  733  	AIDEBUGINFO di;

windows\src\engine\keyman32\appint\aiTIP.h:
  108  	virtual BOOL SendActions();   // I4196
  109: 	virtual BOOL QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR fcontext, PWSTR foutput, DWORD_PTR dwExtraFlags);
  110  

windows\src\engine\keyman32\appint\aiWin2000Unicode.cpp:
  168  
  169: BOOL AIWin2000Unicode::QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR fcontext, PWSTR foutput, DWORD_PTR dwExtraFlags)
  170  {

windows\src\engine\keyman32\appint\aiWin2000Unicode.h:
  63  	virtual BOOL SendActions();   // I4196
  64: 	virtual BOOL QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR fcontext, PWSTR foutput, DWORD_PTR dwExtraFlags);
  65  };

windows\src\engine\keyman32\appint\appint.h:
   45  
   46: // QueueDebugInformation ItemTypes
   47  #define QID_BEGIN_UNICODE		0

  216  
  217: 	virtual BOOL QueueDebugInformation(int ItemType, LPGROUP Group, LPKEY Rule, PWSTR fcontext, PWSTR foutput, DWORD_PTR dwExtraFlags) = 0;
  218  	void SetCurrentShiftState(int ShiftFlags) { FShiftFlags = ShiftFlags; }
```